### PR TITLE
Add inventory host edit form

### PR DESCRIPTION
**Pull request by marshmalien**
##### SUMMARY
Issue: https://github.com/ansible/awx/issues/5264
Mockup: https://tower-mockups.testing.ansible.com/patternfly/inventories/inventories-hosts-edit/

This PR adds the nested inventory host edit form. By leveraging `useRouteMatch`, I could determine if the user navigated to the edit form via "/inventories" or "/hosts", and with that information display the relevant form fields, links, and tabs. This removed the need to create a separate `InventoryHostEdit` container and it also allowed me to delete the `InventoryHostAdd` component and tests. 

Screen Shot 2020-01-22 at 11 10 17 AM

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - UI

##### AWX VERSION

```
awx: 9.1.1
```

### CODE REVIEW

File: awx/ui_next/src/screens/Host/Host.jsx [hunk 1]
```diff
@@ -12,217 +19,190 @@ import ContentError from '@components/ContentError';
...
+      setContentError(error);
     } finally {
-      this.setState({ hasContentLoading: false });
+      setHasContentLoading(false);
     }
+  };
+
+  useEffect(() => {
+    loadHost();
+  }, [location]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Host/Host.jsx:65]: We're not reusing / reloading the host in more than one place. I'm not sure there's a need to move the logic out of the `useEffect` block.

File: awx/ui_next/src/screens/Host/shared/HostForm.jsx [hunk 2]
```diff
@@ -20,6 +20,8 @@ function HostForm({ handleSubmit, handleCancel, host, i18n }) {
     host ? host.summary_fields.inventory : ''
   );
 
+  const hostAddMatch = useRouteMatch('/hosts/add');
```
Comment (marshmalien) [inline awx/ui_next/src/screens/Host/shared/HostForm.jsx:23]: We only want to show the inventory lookup if they are located at `/hosts/add`. The HostEdit, Inventory HostAdd, and Inventory HostEdit views should hide the inventory lookup.

File: awx/ui_next/src/screens/Host/Host.test.jsx [hunk 3]
```diff
@@ -7,39 +8,70 @@ import Host from './Host';
...
+  HostsAPI.readDetail.mockResolvedValue({
+    data: { ...mockDetails },
+  });
+
+  afterEach(() => {
+    wrapper.unmount();
+  });
+
+  test('initially renders succesfully', async () => {
+    history = createMemoryHistory({
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Host/Host.test.jsx:24]: Way to go consolidating thee memoryHistory to 1 place.

File: awx/ui_next/src/screens/Host/HostAdd/HostAdd.jsx [hunk 4]
```diff
@@ -1,58 +1,42 @@
...
+  const hostsMatch = useRouteMatch('/hosts');
+  const inventoriesMatch = useRouteMatch('/inventories/inventory/:id/hosts');
+  const url = hostsMatch ? hostsMatch.url : inventoriesMatch.url;
+
+  const handleSubmit = async formData => {
+    const values = {
+      ...formData,
+      inventory: inventoriesMatch
+        ? inventoriesMatch.params.id
+        : formData.inventory,
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Host/HostAdd/HostAdd.jsx:19]: This is super smart!

**Review by jakemcdermott — APPROVED** [review]

File: awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx [hunk 5]
```diff
@@ -1,46 +1,30 @@
...
+function InventoryHosts({ setBreadcrumb, inventory }) {
   return (
     <Switch>
-      <Route
-        key="host-add"
-        path="/inventories/inventory/:id/hosts/add"
-        render={() => <InventoryHostAdd match={match} />}
-      />
-      ,
+      <Route key="host-add" path="/inventories/inventory/:id/hosts/add">
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx:11]: I think we should keep our routing pattern consistent throughout the app and it's something to keep an eye on as we continue to develop as it's probably out of scope for this PR.

**Review by AlexSCorey — APPROVED** [review]

File: awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx (hunk 5 above)
Comment (marshmalien) [reply awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx:11]: Yes, I completely agree that we should be consistent. I debated between whether or not I should update all of the routes, but decided against it - I didn't want to overwhelm the reviewers with a huge diff not related to the issue.

**Pull request merged**